File: src/main/java/com/example/fileservice/library/LogItem.java
package com.example.fileservice.library;

import lombok.Getter;
import lombok.Setter;

import java.io.IOException;
import java.time.LocalDateTime;

@Getter
public class LogItem extends IOException {
    private final String message;
    private final LocalDateTime dateTime;

    @Setter
    private String correlationId;
    @Setter
    private String type;

    public LogItem(String message) {
        this.message = message;
        this.dateTime = LocalDateTime.now();
    }

    @Override
    public String toString() {
        return "[" + dateTime + "] [" + correlationId + "] " + message;
    }
}

File: src/main/kotlin/com/example/fileservice/library/log/LogItem.kt
package com.example.fileservice.library.log

import java.time.LocalDateTime

open class LogItem constructor(
    val message: String
) {
    private val dateTime: LocalDateTime = LocalDateTime.now()

    private var correlationId: String? = null
    var type: String? = null

    override fun toString(): String {
        return "[$dateTime] [$correlationId] $message"
    }
}
